pvmismatch/contrib/xlsio/xlsio/xlsio.py

```diff
@@ -0,0 +1,88 @@
+    cell_pos = pv_mod.cell_pos
+    cell_pos_df = pd.DataFrame(index=['{}_{}'.format(nr_mod, nr) for nr in range(nrows)])
+    b = 0
+    for bypass in cell_pos:
+        c = 0
+        for col in bypass:
+            cell_pos_df['{}_{}_{}'.format(nr_string, b, c)] = [i['idx'] for i in col]
+            c = c+1
+        b = b+1
+    return cell_pos_df
+
+def _create_irrad_df(pv_mod, cell_pos_df):
+    """Create irradiance dataframe of a module in the PV system"""
+    irrad = pd.Series(pv_mod.Ee.flatten())
+    irrad_df = pd.DataFrame(index=cell_pos_df.index, columns=cell_pos_df.columns)
+    for column in cell_pos_df.columns:
+        for row in cell_pos_df.index:
+            cell_index = cell_pos_df.loc[row, column]
+            irrad_df.loc[row, column] = irrad[cell_index]
+    return irrad_df
+
+def _create_temp_df(pv_mod, cell_pos_df):
+    """Create temperature dataframe of a module in the PV system"""
+    temp = pd.Series(pv_mod.Tcell.flatten())
+    temp_df = pd.DataFrame(index=cell_pos_df.index, columns=cell_pos_df.columns)
+    for column in cell_pos_df.columns:
+        for row in cell_pos_df.index:
+            cell_index = cell_pos_df.loc[row, column]
+            temp_df.loc[row, column] = temp[cell_index]
+    return temp_df
+
+def pvmm_system_layout_to_xls(xls_name, pv_sys, nrows):
+    """Write an xls with worksheets of irradiance, cell temperature and cell index"""
+    writer = pd.ExcelWriter(xls_name, engine='xlsxwriter')
+    workbook = writer.book
+    writer.sheets['CellIndexes'] = workbook.add_worksheet('CellIndexes')
+    writer.sheets['Irradiance'] = workbook.add_worksheet('Irradiance')
+    writer.sheets['CellTemp'] = workbook.add_worksheet('CellTemp')
+    s = 0
+    for string in pv_sys.pvstrs:
+        m = 0
+        for module in string.pvmods:
+            cell_pos_df = _create_cell_pos_df(pv_mod=module, nrows=nrows, nr_string=s, nr_mod=m)
+            if s == 0:
+                startcol = 0
+            else:
+                ncols = int(module.numberCells / nrows)
+                startcol = s*(ncols+1)
+            if m == 0:
+                startrow = 0
+            else:
+                startrow = m*(nrows+1)
+            cell_pos_df.to_excel(writer, sheet_name='CellIndexes', startrow=startrow , startcol=startcol) # write cell pos
+            irrad_df = _create_irrad_df(pv_mod=module, cell_pos_df=cell_pos_df)
+            irrad_df.to_excel(writer, sheet_name='Irradiance', startrow=startrow , startcol=startcol) # write irrad
+            temp_df = _create_temp_df(pv_mod=module, cell_pos_df=cell_pos_df)
+            temp_df.to_excel(writer, sheet_name='CellTemp', startrow=startrow , startcol=startcol) # write irrad
+            m = m + 1
+        s = s + 1
+    # formatting the irradiance worksheet
+    writer.sheets['Irradiance'].conditional_format(0, 0, writer.sheets['Irradiance'].dim_rowmax, writer.sheets['Irradiance'].dim_colmax, {'type': '2_color_scale',
```

Comment:
please observe [pep8](https://pep8.org/) and  wrap long lines :)

Reply:
Thanks a lot for looking into it! I will make the changes once I'm back from holiday.

Reply:
Now lines of code are not longer than 100 characters (sometimes comments are a bit). Please let me know if we are really strict with the 79 char rule :). I can make them shorter of course if necessary.

Reply:
Sorry, yes, the 79 character rule is strict. The only exceptions would be comments _EG_ with URL's that just can't be shortened.